### PR TITLE
docs: improve environment variables section

### PR DESCRIPTION
**Pull request by TC-MO**
remove outdated screenshots and instructions
make sure instructions reflect current state
reformat md table to be more readable
change doc structure and logical flow
add admonitions

### CODE REVIEW

**Review by jirimoravcik — APPROVED** [review]

File: sources/platform/actors/development/programming_interface/environment_variables.md [hunk 1]
```diff
@@ -16,59 +16,83 @@ import TabItem from '@theme/TabItem';
...
 ```
 
 </TabItem>
 </Tabs>
 
-For convenience, rather than using environment vars directly, we provide a `Configuration` class
-that allows reading and updating the Actor configuration.
+## Use the `Configuration` class
+
+For more convenient access to Actor configuration, use the `Configuration` class
```
Comment (janbuchar) [inline sources/platform/actors/development/programming_interface/environment_variables.md:95]: The configuration should be a link here https://docs.apify.com/sdk/js/reference/class/Configuration

File: sources/platform/actors/development/programming_interface/environment_variables.md [hunk 2]
```diff
@@ -16,59 +16,83 @@ import TabItem from '@theme/TabItem';
...
+Here's a table of key system environment variables:
+
+| Environment Variable | Description |
+|----------------------|-------------|
+| `ACTOR_ID` | ID of the Actor. |
+| `ACTOR_RUN_ID` | ID of the Actor run. |
+| `ACTOR_BUILD_ID` | ID of the Actor build used in the run. |
+| `ACTOR_BUILD_NUMBER` | Build number of the Actor build used in the run. |
+| `ACTOR_TASK_ID` | ID of the Actor task. Empty if Actor is run outside of any task, e.g. directly using the API. |
+| `ACTOR_EVENTS_WEBSOCKET_URL` | Websocket URL where Actor may listen for events from Actor platform. |
```
Comment (janbuchar) [inline sources/platform/actors/development/programming_interface/environment_variables.md:30]: A link to the events section would be nice here

File: sources/platform/actors/development/programming_interface/environment_variables.md [hunk 3]
```diff
@@ -16,59 +16,83 @@ import TabItem from '@theme/TabItem';
...
+| `ACTOR_DEFAULT_DATASET_ID` | ID of the dataset where you can push the data. |
+| `ACTOR_DEFAULT_KEY_VALUE_STORE_ID` | ID of the key-value store where the Actor's input and output data are stored. |
+| `ACTOR_DEFAULT_REQUEST_QUEUE_ID` | ID of the request queue that stores and handles requests that you enqueue. |
```
Comment (janbuchar) [inline sources/platform/actors/development/programming_interface/environment_variables.md:33]: Well, yeah, but you can push to any dataset/whatever in your account based on name or ID. This should mention that these are created specifically for a single run.

File: sources/platform/actors/development/programming_interface/environment_variables.md [hunk 4]
```diff
@@ -16,59 +16,83 @@ import TabItem from '@theme/TabItem';
...
+| `ACTOR_INPUT_KEY` | Key of the record in the default key-value store that holds the Actor input. Typically **INPUT**, but may vary. |
```
Comment (janbuchar) [inline sources/platform/actors/development/programming_interface/environment_variables.md:34]: No need to mention the default value here. A link explaining what an input actually is would be helpful.

File: sources/platform/actors/development/programming_interface/environment_variables.md [hunk 5]
```diff
@@ -16,59 +16,83 @@ import TabItem from '@theme/TabItem';
...
+| `ACTOR_MAX_PAID_DATASET_ITEMS` | For paid-per-result Actors, the user-set limit on returned results. Do not exceed this limit. |
+| `APIFY_HEADLESS` | If **1**, web browsers inside the Actor should run in headless mode (no windowing system available). |
+| `APIFY_IS_AT_HOME` | Set to **1** if the Actor is running on Apify servers. |
```
Comment (janbuchar) [inline sources/platform/actors/development/programming_interface/environment_variables.md:37]:
```suggestion
| `APIFY_IS_AT_HOME` | Contains **1** if the Actor is running on Apify servers. |
```

File: sources/platform/actors/development/programming_interface/environment_variables.md [hunk 6]
```diff
@@ -16,59 +16,83 @@ import TabItem from '@theme/TabItem';
...
+| `ACTOR_MEMORY_MBYTES` | Size of memory allocated for the Actor run, in megabytes. Used to optimize memory usage. |
```
Comment (janbuchar) [inline sources/platform/actors/development/programming_interface/environment_variables.md:38]: I'd rather say "Can be used to optimize memory usage or finetuning of low-level external libraries". There's not much that your average actor can do with this information, except maybe for adjusting the size of some cache.

File: sources/platform/actors/development/programming_interface/environment_variables.md [hunk 7]
```diff
@@ -16,59 +16,83 @@ import TabItem from '@theme/TabItem';
...
+| `APIFY_PROXY_PASSWORD` | The Apify Proxy password of the user who started the Actor. |
```
Comment (janbuchar) [inline sources/platform/actors/development/programming_interface/environment_variables.md:39]:
```suggestion
| `APIFY_PROXY_PASSWORD` | A password that can be used to use the Apify Proxy on behalf of the user who started the Actor. |
```

File: sources/platform/actors/development/programming_interface/environment_variables.md [hunk 8]
```diff
@@ -16,59 +16,83 @@ import TabItem from '@theme/TabItem';
...
+| `APIFY_PROXY_PORT` |  |
```
Comment (janbuchar) [inline sources/platform/actors/development/programming_interface/environment_variables.md:40]:
```suggestion
| `APIFY_PROXY_PORT` | TCP port number to be used for connecting to the Apify Proxy  |
```

File: sources/platform/actors/development/programming_interface/environment_variables.md [hunk 9]
```diff
@@ -16,59 +16,83 @@ import TabItem from '@theme/TabItem';
...
+| `APIFY_PROXY_STATUS_URL` |  |
```
Comment (janbuchar) [inline sources/platform/actors/development/programming_interface/environment_variables.md:41]:
```suggestion
| `APIFY_PROXY_STATUS_URL` | URL of a page that contains information about the proxy used to access it. Machine-readable output may be obtained by appending `?format=json` to the URL. |
```

File: sources/platform/actors/development/programming_interface/environment_variables.md [hunk 10]
```diff
@@ -16,59 +16,83 @@ import TabItem from '@theme/TabItem';
...
+| `ACTOR_STANDBY_PORT` | TCP port for the Actor to start an HTTP server to receive messages from Actor Standby. |
```
Comment (janbuchar) [inline sources/platform/actors/development/programming_interface/environment_variables.md:42]:
```suggestion
| `ACTOR_STANDBY_PORT` | TCP port for the Actor to start an HTTP server on to receive messages in the Actor Standby mode. |
```

Also, link to Actor standby docs would be nice here :slightly_smiling_face:

File: sources/platform/actors/development/programming_interface/environment_variables.md [hunk 11]
```diff
@@ -16,59 +16,83 @@ import TabItem from '@theme/TabItem';
...
+| `ACTOR_STARTED_AT` | Date when the Actor was started. |
+| `ACTOR_TIMEOUT_AT` | Date when the Actor will time out. |
+| `APIFY_TOKEN` | API token of the user who started the Actor. |
+| `APIFY_USER_ID` | ID of the user who started the Actor. May differ from the Actor owner. |
+| `ACTOR_WEB_SERVER_PORT` | TCP port for the Actor to start an HTTP server to receive external messages. |
```
Comment (janbuchar) [inline sources/platform/actors/development/programming_interface/environment_variables.md:47]:
```suggestion
| `ACTOR_WEB_SERVER_PORT` | TCP port for the Actor to start an HTTP server on to receive external messages or expose a monitoring or control UI. |
```

File: sources/platform/actors/development/programming_interface/environment_variables.md [hunk 12]
```diff
@@ -16,59 +16,83 @@ import TabItem from '@theme/TabItem';
...
+| `ACTOR_WEB_SERVER_URL` | Unique public URL for accessing the Actor run web server from the outside world. |
+| `APIFY_API_PUBLIC_BASE_URL` | Public URL of the API, typically set to api.apify.com |
```
Comment (janbuchar) [inline sources/platform/actors/development/programming_interface/environment_variables.md:49]:
```suggestion
| `APIFY_API_PUBLIC_BASE_URL` | Public URL of the Apify API, which may be used to interact with the platform. Typically set to api.apify.com |
```

File: sources/platform/actors/development/programming_interface/environment_variables.md [hunk 13]
```diff
@@ -16,59 +16,83 @@ import TabItem from '@theme/TabItem';
...
+| `APIFY_DEDICATED_CPUS` | Number of CPU cores reserved for the actor, based on allocated memory. |
+| `APIFY_DISABLE_OUTDATED_WARNING` |  |
+| `APIFY_FACT` |  |
+| `APIFY_WORKFLOW_KEY` | Identifier used for grouping related runs and API calls together. |
+| `APIFY_META_ORIGIN` |  |
+| `APIFY_SDK_LATEST_VERSION` |  |
```
Comment (janbuchar) [inline sources/platform/actors/development/programming_interface/environment_variables.md:55]:
```suggestion
| `APIFY_SDK_LATEST_VERSION` | Latest available version of the Apify SDK for Javascript, used for checking for updates |
```

File: sources/platform/actors/development/programming_interface/environment_variables.md [hunk 14]
```diff
@@ -16,59 +16,83 @@ import TabItem from '@theme/TabItem';
...
+| `ACTOR_STARTED_AT` | Date when the Actor was started. |
+| `ACTOR_TIMEOUT_AT` | Date when the Actor will time out. |
+| `APIFY_TOKEN` | API token of the user who started the Actor. |
+| `APIFY_USER_ID` | ID of the user who started the Actor. May differ from the Actor owner. |
+| `ACTOR_WEB_SERVER_PORT` | TCP port for the Actor to start an HTTP server to receive external messages. |
+| `ACTOR_WEB_SERVER_URL` | Unique public URL for accessing the Actor run web server from the outside world. |
+| `APIFY_API_PUBLIC_BASE_URL` | Public URL of the API, typically set to api.apify.com |
+| `APIFY_DEDICATED_CPUS` | Number of CPU cores reserved for the actor, based on allocated memory. |
+| `APIFY_DISABLE_OUTDATED_WARNING` |  |
+| `APIFY_FACT` |  |
```
Comment (janbuchar) [inline sources/platform/actors/development/programming_interface/environment_variables.md:52]:
```suggestion
| `APIFY_FACT` | Easter egg |
```

We may just leave this one out I guess :slightly_smiling_face:

File: sources/platform/actors/development/programming_interface/environment_variables.md [hunk 15]
```diff
@@ -16,59 +16,83 @@ import TabItem from '@theme/TabItem';
...
+| `APIFY_WORKFLOW_KEY` | Identifier used for grouping related runs and API calls together. |
+| `APIFY_META_ORIGIN` |  |
```
Comment (janbuchar) [inline sources/platform/actors/development/programming_interface/environment_variables.md:54]: This basically explains how the run was started... possible values here:

image

File: sources/platform/actors/development/programming_interface/environment_variables.md [hunk 16]
```diff
@@ -16,59 +16,83 @@ import TabItem from '@theme/TabItem';
...
+| `ACTOR_STANDBY_PORT` | TCP port for the Actor to start an HTTP server to receive messages from Actor Standby. |
+| `ACTOR_STARTED_AT` | Date when the Actor was started. |
+| `ACTOR_TIMEOUT_AT` | Date when the Actor will time out. |
+| `APIFY_TOKEN` | API token of the user who started the Actor. |
+| `APIFY_USER_ID` | ID of the user who started the Actor. May differ from the Actor owner. |
+| `ACTOR_WEB_SERVER_PORT` | TCP port for the Actor to start an HTTP server to receive external messages. |
+| `ACTOR_WEB_SERVER_URL` | Unique public URL for accessing the Actor run web server from the outside world. |
+| `APIFY_API_PUBLIC_BASE_URL` | Public URL of the API, typically set to api.apify.com |
+| `APIFY_DEDICATED_CPUS` | Number of CPU cores reserved for the actor, based on allocated memory. |
+| `APIFY_DISABLE_OUTDATED_WARNING` |  |
```
Comment (janbuchar) [inline sources/platform/actors/development/programming_interface/environment_variables.md:51]:
```suggestion
| `APIFY_DISABLE_OUTDATED_WARNING` | If you manually set this to `1`, the Apify SDK will not warn you if there is a new version available |
```

**Review by janbuchar — CHANGES_REQUESTED** [review]
Thanks so much for this! Are there still some internal variables that should be included? Also, we may mention the deprecated forms of some variables, for example `APIFY_DEFAULT_DATASET_ID`

File: sources/platform/actors/development/programming_interface/environment_variables.md (hunk 16 above)
Comment (TC-MO) [reply sources/platform/actors/development/programming_interface/environment_variables.md:51]: Is there a default value here? I assume `0`?

File: sources/platform/actors/development/programming_interface/environment_variables.md (hunk 14 above)
Comment (TC-MO) [reply sources/platform/actors/development/programming_interface/environment_variables.md:52]: Best to remove that if nothing of value is added :D

Comment (janbuchar) [reply sources/platform/actors/development/programming_interface/environment_variables.md:52]: Yeah, I think that if somebody can't understand it's an easter egg when they encounter it, they're lost anyway.

**Comment by TC-MO** [comment]
> Thanks so much for this! Are there still some internal variables that should be included? Also, we may mention the deprecated forms of some variables, for example `APIFY_DEFAULT_DATASET_ID`

I kind of hoped the spreadsheet was all that was missing :D

File: sources/platform/actors/development/programming_interface/environment_variables.md (hunk 16 above)
Comment (janbuchar) [reply sources/platform/actors/development/programming_interface/environment_variables.md]: It's an empty string by default

Comment (TC-MO) [reply sources/platform/actors/development/programming_interface/environment_variables.md]: I guess then we don't have to mention default value if it is empty string

**Comment by janbuchar** [comment]
> > Thanks so much for this! Are there still some internal variables that should be included? Also, we may mention the deprecated forms of some variables, for example `APIFY_DEFAULT_DATASET_ID`
> 
> I kind of hoped the spreadsheet was all that was missing :D

If you put everything in the spreadsheet into the docs, then yeah, that should do it :slightly_smiling_face: I don't think that for example `APIFY_INPUT_SECRETS_PRIVATE_KEY_FILE` is in the updated doc though.

File: sources/platform/actors/development/programming_interface/environment_variables.md [hunk 17]
```diff
@@ -16,59 +16,84 @@ import TabItem from '@theme/TabItem';
...
+| `ACTOR_WEB_SERVER_PORT` | TCP port for the Actor to start an HTTP server on. This server can be used to receive external messages or expose monitoring and control interfaces. |
+| `ACTOR_WEB_SERVER_URL` | Unique public URL for accessing the Actor run web server from the outside world. |
+| `APIFY_API_PUBLIC_BASE_URL` | Public URL of the Apify API. May be used to interact with the platform programmatically. Typically set to `api.apify.com`. |
+| `APIFY_DEDICATED_CPUS` | Number of CPU cores reserved for the actor, based on allocated memory. |
+| `APIFY_DISABLE_OUTDATED_WARNING` | Controls the display of outdated version warnings. Set to `1` to suppress notifications about updates. Default value `0`. |
+| `AP IFY_WORKFLOW_KEY` | Identifier used for grouping related runs and API calls together. |
+| `APIFY_META_ORIGIN` | Specifies how an Actor run was started. |
+| `APIFY_SDK_LATEST_VERSION` | Specifies the most recent release version of the Apify SDK for Javascript. Used for checking for updates. |
+| `APIFY_INPUT_SECRETS_KEY_FILE` |  |
+| `APIFY_INPUT_SECRETS_KEY_PASSPHRASE` |  |
```
Comment (janbuchar) [inline sources/platform/actors/development/programming_interface/environment_variables.md:56]:
```suggestion
| `APIFY_INPUT_SECRETS_KEY_FILE` | Path to the secret key that may be used to decrypt [Secret inputs](/platform/actors/development/actor-definition/input-schema/secret-input) |
| `APIFY_INPUT_SECRETS_KEY_PASSPHRASE` | Passphrase for the input secret key mentioned above |
```

File: sources/platform/actors/development/programming_interface/environment_variables.md [hunk 18]
```diff
@@ -16,59 +16,84 @@ import TabItem from '@theme/TabItem';
...
+| `ACTOR_STARTED_AT` | Date when the Actor was started. |
+| `ACTOR_TIMEOUT_AT` | Date when the Actor will time out. |
+| `APIFY_TOKEN` | API token of the user who started the Actor. |
+| `APIFY_USER_ID` | ID of the user who started the Actor. May differ from the Actor owner. |
+| `ACTOR_WEB_SERVER_PORT` | TCP port for the Actor to start an HTTP server on. This server can be used to receive external messages or expose monitoring and control interfaces. |
+| `ACTOR_WEB_SERVER_URL` | Unique public URL for accessing the Actor run web server from the outside world. |
+| `APIFY_API_PUBLIC_BASE_URL` | Public URL of the Apify API. May be used to interact with the platform programmatically. Typically set to `api.apify.com`. |
+| `APIFY_DEDICATED_CPUS` | Number of CPU cores reserved for the actor, based on allocated memory. |
+| `APIFY_DISABLE_OUTDATED_WARNING` | Controls the display of outdated version warnings. Set to `1` to suppress notifications about updates. Default value `0`. |
+| `AP IFY_WORKFLOW_KEY` | Identifier used for grouping related runs and API calls together. |
```
Comment (janbuchar) [inline sources/platform/actors/development/programming_interface/environment_variables.md:52]:
```suggestion
| `APIFY_WORKFLOW_KEY` | Identifier used for grouping related runs and API calls together. |
```

File: sources/platform/actors/development/programming_interface/environment_variables.md [hunk 19]
```diff
@@ -16,59 +16,84 @@ import TabItem from '@theme/TabItem';
...
+| `ACTOR_INPUT_KEY` | Key of the record in the default key-value store that holds the Actor input. Typically **INPUT**, but may vary. |
+| `ACTOR_MAX_PAID_DATASET_ITEMS` | For paid-per-result Actors, the user-set limit on returned results. Do not exceed this limit. |
+| `APIFY_HEADLESS` | If **1**, web browsers inside the Actor should run in headless mode (no windowing system available). |
+| `APIFY_IS_AT_HOME` | Contains **1** if the Actor is running on Apify servers. |
+| `ACTOR_MEMORY_MBYTES` | Size of memory allocated for the Actor run, in megabytes. Can be used to optimize memory usage or finetuning of low-level external libraries. |
+| `APIFY_PROXY_PASSWORD` | Password for accessing Apify Proxy services. This password enables the Actor to utilize nproxy servers on behalf of the user who initiated the Actor run. |
+| `APIFY_PROXY_PORT` | TCP port number to be used for connecting to the Apify Proxy. |
+| `APIFY_PROXY_STATUS_URL` | Endpoint URL for regrieving proxy status information. Appending `?format=json` to this URL returns the data in JSON format for programmatic processing. |
```
Comment (janbuchar) [inline sources/platform/actors/development/programming_interface/environment_variables.md:41]:
```suggestion
| `APIFY_PROXY_STATUS_URL` | URL for retrieving proxy status information. Appending `?format=json` to this URL returns the data in JSON format for programmatic processing. |
```

File: sources/platform/actors/development/programming_interface/environment_variables.md [hunk 20]
```diff
@@ -16,59 +16,84 @@ import TabItem from '@theme/TabItem';
...
+| `ACTOR_STANDBY_PORT` | TCP port for the Actor to start an HTTP server to receive messages in the [Actor Standby](/platform/actors/development/programming-interface/standby) mode. |
+| `ACTOR_STARTED_AT` | Date when the Actor was started. |
+| `ACTOR_TIMEOUT_AT` | Date when the Actor will time out. |
+| `APIFY_TOKEN` | API token of the user who started the Actor. |
+| `APIFY_USER_ID` | ID of the user who started the Actor. May differ from the Actor owner. |
+| `ACTOR_WEB_SERVER_PORT` | TCP port for the Actor to start an HTTP server on. This server can be used to receive external messages or expose monitoring and control interfaces. |
+| `ACTOR_WEB_SERVER_URL` | Unique public URL for accessing the Actor run web server from the outside world. |
+| `APIFY_API_PUBLIC_BASE_URL` | Public URL of the Apify API. May be used to interact with the platform programmatically. Typically set to `api.apify.com`. |
+| `APIFY_DEDICATED_CPUS` | Number of CPU cores reserved for the actor, based on allocated memory. |
+| `APIFY_DISABLE_OUTDATED_WARNING` | Controls the display of outdated version warnings. Set to `1` to suppress notifications about updates. Default value `0`. |
```
Comment (janbuchar) [inline sources/platform/actors/development/programming_interface/environment_variables.md:51]:
```suggestion
| `APIFY_DISABLE_OUTDATED_WARNING` | Controls the display of outdated version warnings. Set to `1` to suppress notifications about updates. |
```

**Review by janbuchar — CHANGES_REQUESTED** [review]

File: sources/platform/actors/development/programming_interface/environment_variables.md (hunk 3 above)
Comment (TC-MO) [reply sources/platform/actors/development/programming_interface/environment_variables.md]: rephrased description to mention that these are run specific values

File: sources/platform/actors/development/programming_interface/environment_variables.md (hunk 4 above)
Comment (TC-MO) [reply sources/platform/actors/development/programming_interface/environment_variables.md]: added link to input section of docs

File: sources/platform/actors/development/programming_interface/environment_variables.md (hunk 15 above)
Comment (TC-MO) [reply sources/platform/actors/development/programming_interface/environment_variables.md]: added link with origin section

**Review by janbuchar — APPROVED** [review]
Good stuff!

**Pull request merged**